frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segment_list_item.tsx

```diff
@@ -230,14 +234,17 @@ function _MeshInfoItem(props: {
   setAdditionalCoordinates: (additionalCoordinates: AdditionalCoordinate[] | undefined) => void;
 }) {
   const dispatch = useDispatch();
-
   const onChangeMeshVisibility = (layerName: string, id: number, isVisible: boolean) => {
-    dispatch(updateMeshVisibilityAction(layerName, id, isVisible));
+    dispatch(updateMeshVisibilityAction(layerName, id, isVisible, mesh?.seedAdditionalCoordinates));
   };
 
   const { segment, isSelectedInList, isHovered, mesh } = props;
 
-  if (!mesh) {
+  if (
+    !mesh ||
+    getAdditionalCoordinatesAsString(mesh.seedAdditionalCoordinates || null) !==
+      getAdditionalCoordinatesAsString(Store.getState().flycam.additionalCoordinates)
```

Comment:
code that renders JSX should not access `Store.getState()`. Please use `useSelector((state: OxalisState) => state.flycam....` at the top of the rendering-function instead.